MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs

```diff
@@ -1155,29 +1158,78 @@ public ReadOnlySpan<byte> Invoke(ushort ordinal, bool offsetsOnly = false)
         }
 
         /// <summary>
-        ///     Sets up the Generic MajorBBS Btrieve Database for Use
+        ///     Sets up the Generic MajorBBS Btrieve Database for use
         /// </summary>
         private void SetupGENBB()
         {
-            //If it's already setup, bail
-            if (Module.Memory.TryGetVariablePointer("GENBB", out _))
-                return;
+            //Setup the Pointer to the Global GENBB Address -- ensuring each module is referencing the same GENBB Pointer & GENBB Processor
+            if (!_globalCache.TryGet<IntPtr16>("GENBB-POINTER", out var genbbPointer))
+            {
+                //Declare Pointers and Locations for GENBB Data
+                Module.Memory.AddSegment(0x1000);
+                genbbPointer = new IntPtr16(0x1000, 0x0); //GENBB Btrieve Struct
+                var genbbNamePointer = new IntPtr16(0x1000, 0x100); //GENBB File Name Pointer
+                var genbbDataPointer = new IntPtr16(0x1000, 0x200); //GENBB Record Pointer
+
+                //Set Struct Value
+                var newBtvStruct = new BtvFileStruct { filenam = genbbNamePointer, reclen = 8192, data = genbbDataPointer };
+                Module.Memory.SetArray(genbbPointer, newBtvStruct.Data);
 
-            var btvFileStructPointer = Module.Memory.AllocateVariable("BBSGEN-STRUCT", BtvFileStruct.Size);
-            var btvFileName = Module.Memory.AllocateVariable("BBSGEN-NAME", 11); //BBSGEN.DAT\0
-            Module.Memory.SetArray("BBSGEN-NAME", Encoding.ASCII.GetBytes($"BBSGEN.DAT\0"));
-            var btvDataPointer =
-                Module.Memory.AllocateVariable("BBSGEN-POINTER", 8192); //GENSIZ -- Defined in MAJORBBS.H
+                //Set Filename Value
+                Module.Memory.SetArray(genbbNamePointer, Encoding.ASCII.GetBytes($"BBSGEN.DAT\0"));
 
-            var newBtvStruct = new BtvFileStruct { filenam = btvFileName, reclen = 8192, data = btvDataPointer };
+                //Set Global Cache Values
+                _globalCache.Set("GENBB-POINTER", genbbPointer);
+                _globalCache.Set("GENBB-PROCESSOR",
+                    new BtrieveFileProcessor(_fileFinder, "BBSGEN.DAT", Directory.GetCurrentDirectory( )));
+            }
```

Comment:
how about moving this to a helper function?

Reply:
This is already in the helper function `SetupGENBB` (and similar `SetupACCBB`). How would you advise how to break this out further?

Reply:
It seems very similar to SetupGENBB/SetupACCBB. Can these methods be merged together/parameterized?

Reply:
Consolidated them into one method in the base class. I had to add a little logic around not re-declaring the processor if one already exists, because the `ACCBB` processor is setup in Program.cs because it's needed by the Database Reset.